src/marqo/tensor_search/tensor_search.py

```diff
@@ -934,12 +943,14 @@ def _get_documents_for_upsert(
                 dummy_res = result
                 break
 
-        # Put the chunks list in res_data, so it's complete
+        # Put the chunks list in res_data, so it contains all doc data
         if doc_in_results:
-            res_data["_source"]["__chunks"] = res_chunks["_source"]["__chunks"]
+            # Only add chunks if not a chunkless doc
+            if res_chunks["_source"]:
+                res_data["_source"]["__chunks"] = res_chunks["_source"]["__chunks"]
```

Comment:
are there any assumptions being made here? For example, are we guaranteed that `res_chunks["_source"] != {}` implies `"__chunks" in res_chunks["_source"]`

Reply:
The concern is a possible KeyError

Reply:
Throughout my testing, yes, if `res_chunks["_source"]` is non-empty, it must have the `"__chunks"` key. this is because in its query, it only includes things inside the `"__chunks"` key.

if you want it to be assumptionless, I suppose
```
if "__chunks" in res_chunks["_source"]:
      res_data["_source"]["__chunks"] = res_chunks["_source"]["__chunks"]
```

would do it

Reply:
That makes sense, thanks for the explanation